packages/lisk-p2p/src/peer.ts

```diff
@@ -97,17 +97,18 @@ export interface PeerConnectionState {
 export const constructPeerId = (ipAddress: string, wsPort: number): string =>
 	`${ipAddress}:${wsPort}`;
 
-export const constructPeerIdFromPeerInfo = (peerInfo: P2PPeerInfo): string =>
-	`${peerInfo.ipAddress}:${peerInfo.wsPort}`;
+export const constructPeerIdFromPeerInfo = (
+	peerInfo: P2PPeerInfo | P2PDiscoveredPeerInfo,
+): string => `${peerInfo.ipAddress}:${peerInfo.wsPort}`;
```

Comment:
`P2PDiscoveredPeerInfo` inherits from `P2PPeerInfo`, so you shouldn't need the `|`. The argument type could just be `P2PPeerInfo`.

Reply:
Yeah I was just trying to be specific, you are right we can only use `P2PPeerInfo`